cmd/snapd-generator/main.c

```diff
@@ -97,3 +90,80 @@ int main(int argc, char **argv)
 	fprintf(f, "WantedBy=local-fs.target\n");
 	return 0;
 }
+
+int ensure_fusesquashfs_inside_container(const char *normal_dir)
+{
+	// check if we are running inside a container
+	if (system("systemd-detect-virt --container --quiet") != 0) {
+		return 0;
+	}
+
+	char *fstype = NULL;
+	if (system("which squashfuse > /dev/null") == 0) {
```

Comment:
stderr should be redirected to dev/null too. some implementations of which print stuff to stdout and some to stderr

Reply:
No longer relevant due to custom path lookup.